app/javascript/widget/mixins/nextAvailabilityTime.js

```diff
@@ -0,0 +1,256 @@
+          nextWorkingHour = this.getNextWorkingHour(nextDay);
+        }
+        return nextWorkingHour;
+      }
+      return null;
+    },
+    newDateWithTimeZone() {
+      const date = new Date();
+      const timeZone = this.timeZoneValue;
+      const zonedDate = utcToZonedTime(date, timeZone);
+      return zonedDate;
+    },
+    presentHour() {
+      return this.newDateWithTimeZone.getHours();
+    },
+    presentMinute() {
+      return this.newDateWithTimeZone.getMinutes();
+    },
+    currentDay() {
+      const date = this.newDateWithTimeZone;
+      const day = date.getDay();
+      const currentDay = Object.keys(this.dayNames).find(
+        key => this.dayNames[key] === this.dayNames[day]
+      );
+      return Number(currentDay);
+    },
+    currentDayOpenHour() {
+      return this.workingHoursEnabled
+        ? this.currentDayWorkingHours.open_hour
+        : null;
+    },
+    currentDayOpenMinute() {
+      return this.workingHoursEnabled
+        ? this.currentDayWorkingHours.open_minutes
+        : null;
+    },
+    currentDayCloseHour() {
+      return this.workingHoursEnabled
+        ? this.currentDayWorkingHours.close_hour
+        : null;
+    },
+    nextDayOpenHour() {
+      if (this.workingHoursEnabled) {
+        return this.nextDayWorkingHours.open_all_day
+          ? 0
+          : this.nextDayWorkingHours.open_hour;
+      }
+      return null;
+    },
+    nextDayOpenMinute() {
+      return this.workingHoursEnabled
+        ? this.nextDayWorkingHours.open_minutes
+        : null;
+    },
+    dayDiff() {
+      // Here this is used to get the difference between current day and next working day
+      if (this.workingHoursEnabled) {
+        const currentDay = this.currentDay;
+        const nextDay = this.nextDayWorkingHours.day_of_week;
+        const totalDays = 6;
+        return nextDay > currentDay
+          ? nextDay - currentDay
+          : totalDays - currentDay + nextDay;
+      }
+      return null;
+    },
+    dayNameOfNextWorkingDay() {
+      if (this.workingHoursEnabled) {
+        return this.dayNames[this.nextDayWorkingHours.day_of_week];
+      }
+      return null;
+    },
+    hoursAndMinutesBackInOnline() {
+      if (this.presentHour >= this.currentDayCloseHour) {
+        return this.getHoursAndMinutesUntilNextDayOpen(
+          this.nextDayOpenHour,
+          this.nextDayOpenMinute,
+          this.currentDayCloseHour
+        );
+      }
+      return this.getHoursAndMinutesUntilNextDayOpen(
+        this.currentDayOpenHour,
+        this.currentDayOpenMinute,
+        this.currentDayCloseHour
+      );
+    },
+    exactTimeInAmPm() {
+      return `${this.timeSlot.day !== this.currentDay ? 'tomorrow' : ''} ${
+        this.timeSlot.day !== this.currentDay ? '' : `at ${this.timeSlot.from}`
+      }`;
+    },
+    hoursLeftValue() {
+      const {
+        hours: hoursLeft,
+        minutes: minutesLeft,
+      } = this.hoursAndMinutesBackInOnline;
```

Comment:
Since we control the implementation of `hoursAndMinutesBackInOnline` we can send the values with `{ hoursLeft: 0, minutesLeft: 45}` directly instead of `hours` and `minutes` as keys. We will avoid the destructuring assignment